user/utils.py

```diff
@@ -0,0 +1,25 @@
+import json, bcrypt, jwt
+
+from django.http import JsonResponse
+
+from user.models import User
+from my_settings import SECRET_KEY, ALGORITHM
+
+
+def login_decorator(func):
+    def wrapper(self, request, *args, **kwargs):
+        try:
+            auth_token   = request.headers.get('Authorization', None)
+            payload      = jwt.decode(auth_token, SECRET_KEY, algorithm = ALGORITHM)
+            user         = User.objects.get(id=payload['id'])
+            request.user = user
+
+            return func(self, request, *args, **kwargs)
+
+        except jwt.DecodeError:
+            return JsonResponse({'message':'UNVALID_TOKEN'}, status=403)
+
+        except User.DoesNotExist:
+            return JsonResponse({'message':'UNVALID_USER'}, status=401)
```

Comment:
```suggestion
        except jwt.exceptions.DecodeError:
            return JsonResponse({'message':'INVALID_TOKEN'}, status=403)

        except User.DoesNotExist:
            return JsonResponse({'message':'INVALID_USER'}, status=401)
```
- `DecodeError` 잡아주시는건 좋은데 이 부분 잘 잡히는지 테스트 해보신건가요? 왜냐하면 `DecodeError` 는 `jwt.exceptions` 내에 존재하는 익셉션 클래스라서 `jwt.exceptions.DecodeError` 로 잡아주셔야 정상작동할텐데요. 확인해주세요.

Reply:
수정했습니다